src/backend/app/tasks/tasks_crud.py

```diff
@@ -185,15 +184,34 @@ async def create_task_history_for_status_change(
     )
     log.info(msg)
 
-    new_task_history = db_models.DbTaskHistory(
-        project_id=db_task.project_id,
-        task_id=db_task.id,
-        action=get_action_for_status_change(new_status),
-        action_text=msg,
-        actioned_by=db_user,
-        user_id=db_user.id,
+    query = text(
+        """
+        INSERT INTO task_history (
+            project_id, task_id, action, action_text,
+            action_date, user_id
+        )
+        VALUES (
+            :project_id, :task_id, :action, :action_text,
+            :action_date, :user_id
+        )
+        RETURNING *
+    """
     )
 
+    params = {
+        "project_id": db_task.project_id,
+        "task_id": db_task.id,
+        "action": get_action_for_status_change(new_status).name,
+        "action_text": msg,
+        "action_date": datetime.now(),
+        "user_id": db_user.id,
+    }
+
+    result = db.execute(query, params)
+    db.commit()
+
+    row = result.fetchone()
```

Comment:
instead of `fetchone() `simply use` first()`, `first()` will call only first row all the time, while `fetchone()` returns all the rows but only one at a time.

Reply:
This is true for SQLAlchemy, but not for psycopg actually!

Psycopg only has fetchone fetchmany and fetchall 😄

Apologies for not mentioning that!

Reply:
Actually you are technically correct!

I noticed this PR is using SQLAlchemy! Commented in the main thread